api/capi/src/nntrainer.cpp

```diff
@@ -354,21 +363,67 @@ int ml_train_model_run(ml_train_model_h model, ...) {
   while ((data = va_arg(arguments, const char *))) {
     arg_list.push_back(data);
   }
-
   va_end(arguments);
 
+  return nntrainer_model_compile(model, arg_list);
+}
+
+static int nntrainer_model_run(ml_train_model_h model,
+                               std::vector<std::string> &param_list) {
+  int status = ML_ERROR_NONE;
+  ml_train_model *nnmodel;
+  std::shared_ptr<ml::train::Model> m;
+
+  ML_TRAIN_VERIFY_VALID_HANDLE(model);
+
   {
     ML_TRAIN_GET_VALID_MODEL_LOCKED(nnmodel, model);
     ML_TRAIN_ADOPT_LOCK(nnmodel, model_lock);
     m = nnmodel->model;
   }
 
-  returnable f = [&]() { return m->train(arg_list); };
+  returnable f = [&]() { return m->train(param_list); };
   status = nntrainer_exception_boundary(f);
 
   return status;
 }
 
+int ml_train_model_run_with_params(ml_train_model_h model, const char *params) {
```

Comment:
```ml_train_layer_set_property``` also has valist. Do we need to do the same way?

Reply:
yes